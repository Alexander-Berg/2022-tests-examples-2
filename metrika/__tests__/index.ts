import { remap } from '../normalize';

describe('remap', () => {
    it('works', () => {
        const result = remap(
            { yandexEmployeeLogin: '13' },
            { 13: 'loginFromStaff' },
        );

        expect(result).toEqual({ yandexEmployeeLogin: 'loginFromStaff' });
    });

    it('skips keys if they not presented in raw object', () => {
        const result = remap({ yandexEmployeeLogin: '13' }, {});

        expect(result).toEqual({});
    });

    it('skips keys if they not presented in mapping object', () => {
        const result = remap({}, { 13: 'loginFromStaff' });

        expect(result).toEqual({});
    });

    it('mixed case', () => {
        const result = remap(
            { someOtherField: '42' },
            { 13: 'loginFromStaff' },
        );

        expect(result).toEqual({});
    });
});
