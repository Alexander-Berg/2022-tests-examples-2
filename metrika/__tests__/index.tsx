import * as React from 'react';

import { render, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { Highlight, Substr } from '..';
configure({ adapter: new Adapter() });

describe('Highlight', () => {
    it('работает по строке', () => {
        const container = render(
            <div>
                <Highlight pattern="екс">текст</Highlight>
            </div>,
        );

        expect(container).toMatchSnapshot();
    });
    it('работает с вложенными строками', () => {
        const container = render(
            <div>
                <Highlight pattern="екс">
                    <span>текст</span>
                </Highlight>
            </div>,
        );

        expect(container).toMatchSnapshot();
    });
    it('работает с несколькими вложенными строками', () => {
        const container = render(
            <div>
                <Highlight pattern="екс">
                    <span>текст</span>
                    <span>текст</span>
                </Highlight>
            </div>,
        );

        expect(container).toMatchSnapshot();
    });
    it('работает с несколькими паттернами через пробел', () => {
        const container = render(
            <div>
                <Highlight pattern="ако екс">
                    <span>какой-то текст</span>
                    <span>какой-то текст</span>
                </Highlight>
            </div>,
        );

        expect(container).toMatchSnapshot();
    });
    it('работает с паттернами через массив', () => {
        const container = render(
            <div>
                <Highlight pattern={['ако', 'екс']}>
                    <span>какой-то текст</span>
                    <span>какой-то текст</span>
                </Highlight>
            </div>,
        );

        expect(container).toMatchSnapshot();
    });
    it('работает с подсветкой фона', () => {
        const container = render(
            <div>
                <Highlight pattern={['ако', 'екс']} highlight="background">
                    <span>какой-то текст</span>
                    <span>какой-то текст</span>
                </Highlight>
            </div>,
        );

        expect(container).toMatchSnapshot();
    });
    it('работает позиционная подсветка', () => {
        const container = render(
            <div>
                <Substr match={{ start: 1, end: 3 }}>Here comes Johny!</Substr>
            </div>,
        );

        expect(container).toMatchSnapshot();
    });
});
