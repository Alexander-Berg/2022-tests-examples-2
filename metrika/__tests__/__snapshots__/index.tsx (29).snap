// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`float negative 1`] = `
.c0 {
  white-space: nowrap;
}

<span
  class="c0"
>
  -10
</span>
`;

exports[`float null 1`] = `
.c0 {
  white-space: nowrap;
}

<span
  class="c0"
>
  н/д
</span>
`;

exports[`float positive 1`] = `
.c0 {
  white-space: nowrap;
}

<span
  class="c0"
>
  10
</span>
`;

exports[`float too high 1`] = `
.c0 {
  white-space: nowrap;
}

<span
  class="c0"
>
  10
  <sup>
    6
  </sup>
</span>
`;

exports[`float too low 1`] = `
.c0 {
  white-space: nowrap;
}

<span
  class="c0"
>
  &lt; 0,01
</span>
`;

exports[`float too low negative 1`] = `
.c0 {
  white-space: nowrap;
}

<span
  class="c0"
>
  &lt; 0,01
</span>
`;

exports[`float undefined 1`] = `
.c0 {
  white-space: nowrap;
}

<span
  class="c0"
>
  н/д
</span>
`;

exports[`int negative 1`] = `
.c0 {
  white-space: nowrap;
}

<span
  class="c0"
>
  -10
</span>
`;

exports[`int null 1`] = `
.c0 {
  white-space: nowrap;
}

<span
  class="c0"
>
  н/д
</span>
`;

exports[`int positive 1`] = `
.c0 {
  white-space: nowrap;
}

<span
  class="c0"
>
  10
</span>
`;

exports[`int too high 1`] = `
.c0 {
  white-space: nowrap;
}

<span
  class="c0"
>
  1 000 000
</span>
`;

exports[`int too high compress 1`] = `
.c0 {
  white-space: nowrap;
}

<span
  class="c0"
>
  1
   
  млн
</span>
`;

exports[`int too low 1`] = `
.c0 {
  white-space: nowrap;
}

<span
  class="c0"
>
  0
</span>
`;

exports[`int undefined 1`] = `
.c0 {
  white-space: nowrap;
}

<span
  class="c0"
>
  н/д
</span>
`;

exports[`percent negative 1`] = `
.c0 {
  white-space: nowrap;
}

<span
  class="c0"
>
  -10
   
  <span>
    %
  </span>
</span>
`;

exports[`percent null 1`] = `
.c0 {
  white-space: nowrap;
}

<span
  class="c0"
>
  н/д
</span>
`;

exports[`percent positive 1`] = `
.c0 {
  white-space: nowrap;
}

<span
  class="c0"
>
  10
   
  <span>
    %
  </span>
</span>
`;

exports[`percent too high 1`] = `
.c0 {
  white-space: nowrap;
}

<span
  class="c0"
>
  10
  <sup>
    6
  </sup>
   
  <span>
    %
  </span>
</span>
`;

exports[`percent too low 1`] = `
.c0 {
  white-space: nowrap;
}

<span
  class="c0"
>
  &lt; 0,01
   
  <span>
    %
  </span>
</span>
`;

exports[`percent undefined 1`] = `
.c0 {
  white-space: nowrap;
}

<span
  class="c0"
>
  н/д
</span>
`;
